torch/lib/THPP/tensors/generic/THTensor.cpp

```diff
@@ -7,7 +7,7 @@
 #define const_storage_cast(storage) \
   dynamic_cast<const THStorage<real>&>(storage)
 #define const_long_cast(tensor) \
-  dynamic_cast<const THTensor<std::conditional<sizeof(long) == 8, long, int64_t>::type>&>(tensor)
```

Comment:
@peterjc123 what's the reason for adding these templates? they seem to break the build on OS X

Reply:
Just to use int64_t cast for MSVC. The long type takes up 4 bytes using MSVC while 8 bytes using GCC. If it fails to build, the following macro could be used as a second choice.
```Cpp
#ifndef _WIN32
#define const_long_cast(tensor) \
   dynamic_cast<const THTensor<long>::type>&>(tensor)
#else
#define const_long_cast(tensor) \
   dynamic_cast<const THTensor<int64_t>::type>&>(tensor)
#endif
```

Reply:
It was originally introduced for consistency. @fmassa don't know the difference between long long and long in THPP. So i just keep using long for Unix systems here.